lib/webtask.js

```diff
@@ -708,6 +760,34 @@ exports.ensure_authenticated = function(ctx) {
             });
         }
 
+        function validate_authentication_token(secret, expected_aud) {
+            var auth_token;
+            var decoded;
+
+            if(req.headers['authorization']) {
+                try {
+                    auth_token = req.headers['authorization'].match(/^bearer\s+(.+)$/i)[1];
+                } catch(e) {
+                    throw { code: 400, message: 'could not parse supplied Authorization header' + req.headers['authorization'] + ':' + e.message, req_id: req.req_id };
+                }
+            } else {
+                auth_token = req.query.key;
+            }
+
+            decoded = jws.decode(auth_token);
+
+            if(!jws.verify(auth_token, decoded.header.alg, secret))
```

Comment:
Actually I see this is caught in the outer scope, but you probably want to control the status code etc in this case, so explicitly testing this condition would be required.